### PR TITLE
feat: add ADR-023 for cross-model reviewer assignment

### LINKED ISSUE #168: R19: Cross-model validation for high-risk changes
## Summary

When multi-model support becomes available, run one reviewer on a different model family for security-sensitive changes. Research shows heterogeneous models (different vendors) improve adversarial review results, while same-model debate does not.

## Research Justification

- [Multi-Agent Debate](https://d2jud02ci9yv69.cloudfront.net/2025-04-28-mad-159/blog/mad/) (ICLR/NeurIPS 2025): "MAD consistently underperforms simpler approaches" with same-model agents. **Exception: heterogeneous models (different vendors) showed improvement.** Devil's advocate pattern performed worst with same model. Token consumption 2-3x higher without quality gains.
- [Rise of AI Teammates in SE 3.0](https://arxiv.org/html/2507.15003v1): "Cross-vendor review provides stronger diversity than same-vendor adversarial prompting."
- Current state: All dev-team agents run on the same model family (Claude). For most changes this is fine, but security-sensitive changes would benefit from cross-model diversity.

## Implementation Approach

1. **Prerequisite**: Multi-model support in Claude Code or dev-team's agent spawning
2. For changes tagged as security-sensitive (auth, crypto, session management):
   - Run Szabo on the primary model (Claude)
   - Run a second security reviewer on an alternative model family
   - Compare findings — agreements are high-confidence, disagreements warrant human attention
3. Model selection configurable in project config
4. Fallback: if alternative model unavailable, proceed with single-model review (current behavior)

## Acceptance Criteria

- [ ] Architecture supports spawning agents on different model families
- [ ] Security-sensitive changes trigger cross-model review
- [ ] Agreement/disagreement analysis between models
- [ ] Configurable model selection
- [ ] Graceful fallback to single-model review
- [ ] ADR documenting the cross-model validation pattern

## References

- Research synthesis: docs/research-synthesis.md (Section 3: Adversarial Code Review, R19)
- Priority: P3 — blocked on multi-model support availability
- This is a future capability; the issue is created now for tracking and design thinking

### PR DESCRIPTION
**Pull request by fredericboyer**
## Summary

- Add ADR-023 documenting the decision to support cross-model reviewer assignment when multi-model runtime becomes available
- Extend reviewer agent definitions (Szabo, Knuth, Brooks) with `cross_model` field in both `.dev-team/agents/` and `templates/agents/`
- Research-backed: MAD (ICLR/NeurIPS 2025) found heterogeneous models improve adversarial review while same-model debate does not

## Details

**Current state**: All agents run on the same model family (Claude), differentiated only by tier (Opus/Sonnet per ADR-008).

**Future state**: Reviewer agents can be assigned alternative models via `cross_model` field for high-risk changes (security, architectural boundaries, DEEP review depth).

**Status**: Aspirational — blocked on multi-model runtime support in Claude Code. The `cross_model` field is ignored by current runtimes but documents intent.

Closes #168

## Test plan

- [x] All 308 tests pass (`npm test`)
- [ ] Verify ADR-023 is indexed in `docs/adr/README.md`
- [ ] Verify `cross_model` field present in all 6 reviewer agent definitions
- [ ] Confirm no behavioral change — field is documentation-only

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: templates/agents/dev-team-szabo.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: dev-team-szabo
 description: Security auditor. Use to review code for vulnerabilities, audit auth flows, analyze attack surfaces, and assess dependency risks. Read-only — does not modify code.
 tools: Read, Grep, Glob, Bash, Agent
 model: opus
+cross_model: # recommended when multi-model runtime available (ADR-023)
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply templates/agents/dev-team-szabo.md]: Acknowledged. The `cross_model` field will be removed entirely from agent definitions in the v1.0.0 release PR, since multi-model runtime is not yet available. This addresses the type mismatch (boolean vs string per ADR-023).

File: docs/adr/023-cross-model-reviewer-assignment.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Implementation tasks (code generation should remain single-model for consistency)
+- Documentation-only changes
+
+### Activation mechanism
+
+When multi-model runtime is available, the orchestrator (Drucker) can request cross-model validation by passing the `cross_model` preference when spawning a reviewer agent. The runtime resolves the model identifier to an available provider. If the cross-model provider is unavailable, the agent falls back to its primary `model` assignment silently.
+
+## Consequences
+
+- Reviewer agent definitions gain a `cross_model` field documenting the recommended alternative model — this serves as documentation of intent even before runtime support exists
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply docs/adr/023-cross-model-reviewer-assignment.md]: Fixed in release PR #247: Removed the `cross_model` field entirely from all 6 agent definitions (3 templates + 3 local). Per ADR-023, the field should only be present when set to a vendor-qualified model identifier string. Since multi-model runtime is not yet available, the field is omitted rather than set to false or null.

File: templates/agents/dev-team-szabo.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fredericboyer) [reply templates/agents/dev-team-szabo.md]: Fixed in release PR #247: Removed the `cross_model` field entirely from all 6 agent definitions (3 templates + 3 local). Per ADR-023, the field should only be present when set to a vendor-qualified model identifier string. Since multi-model runtime is not yet available, the field is omitted rather than set to false or null.

File: templates/agents/dev-team-knuth.md [hunk 3]
```diff
@@ -3,6 +3,7 @@ name: dev-team-knuth
 description: Quality auditor. Use to audit code for correctness gaps, missing test coverage, boundary conditions, and unproven assumptions. Read-only — identifies what is missing or unproven, does not write code or tests.
 tools: Read, Grep, Glob, Bash, Agent
 model: opus
+cross_model: # recommended when multi-model runtime available (ADR-023)
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply templates/agents/dev-team-knuth.md]: Fixed in release PR #247: Removed the `cross_model` field entirely from all 6 agent definitions (3 templates + 3 local). Per ADR-023, the field should only be present when set to a vendor-qualified model identifier string. Since multi-model runtime is not yet available, the field is omitted rather than set to false or null.

File: .dev-team/agents/dev-team-knuth.md [hunk 4]
```diff
@@ -3,6 +3,7 @@ name: dev-team-knuth
 description: Quality auditor. Use to audit code for correctness gaps, missing test coverage, boundary conditions, and unproven assumptions. Read-only — identifies what is missing or unproven, does not write code or tests.
 tools: Read, Grep, Glob, Bash, Agent
 model: opus
+cross_model: # recommended when multi-model runtime available (ADR-023)
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply .dev-team/agents/dev-team-knuth.md]: Fixed in release PR #247: Removed the `cross_model` field entirely from all 6 agent definitions (3 templates + 3 local). Per ADR-023, the field should only be present when set to a vendor-qualified model identifier string. Since multi-model runtime is not yet available, the field is omitted rather than set to false or null.

File: .dev-team/agents/dev-team-brooks.md [hunk 5]
```diff
@@ -3,6 +3,7 @@ name: dev-team-brooks
 description: Architect and quality attribute reviewer. Use to review architectural decisions, challenge coupling and dependency direction, validate changes against ADRs, and assess quality attributes (performance, maintainability, scalability). Always-on for all non-test code changes. Read-only — does not modify code.
 tools: Read, Grep, Glob, Bash, Agent
 model: opus
+cross_model: # recommended when multi-model runtime available (ADR-023)
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply .dev-team/agents/dev-team-brooks.md]: Fixed in release PR #247: Removed the `cross_model` field entirely from all 6 agent definitions (3 templates + 3 local). Per ADR-023, the field should only be present when set to a vendor-qualified model identifier string. Since multi-model runtime is not yet available, the field is omitted rather than set to false or null.
